### PR TITLE
Add Antora Build to docs-build branch

### PR DESCRIPTION
**Pull request by mojavelinux**
This PR populates the Antora playbook branch for building the production docs site using Antora from Gradle.

The Antora playbook branch, named docs-build, will take over the responsibility of deploying the production docs. That means it will contain the Antora playbook file for the production docs (antora-playbook.yml) as well as the Deploy Docs and Rebuild Search Index CI workflows. The main branch will still continue to be used as the scheduler for these workflows.

There will be a corresponding PR to the main branch to set up parallel workflows to be used as the build scheduler and trigger for the workflows in this branch. That PR will also repurpose the Gradle build in the docs folder to only build docs in that branch.

This PR contains an Antora playbook for building the production docs site from the current repository, considering any linked worktrees. That playbook is intended for testing the full site build.

### CODE REVIEW

File: lib/antora/extensions/antora-linked-worktree-patch.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+'use strict'
+
+/* Copyright (c) 2022 OpenDevise, Inc.
```
Comment (rwinch) [inline lib/antora/extensions/antora-linked-worktree-patch.js:3]: Sorry I missed this in your initial posting, but given this is hosted in Spring Security's code base can we please get the Copyright updated to be:

>  Copyright 2002-2022 the original author or authors.

File: lib/antora/extensions/antora-linked-worktree-patch.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License Version 2.0. If a copy of this license was not distributed
+ * with this file, you can obtain one at http://mozilla.org/MPL/2.0/.
```
Comment (rwinch) [inline lib/antora/extensions/antora-linked-worktree-patch.js:7]: Again sorry I missed this in your initial posting, but given this is hosted in Spring Security's code base can we please get this updated to Apache 2 license header?

**Review by rwinch — CHANGES_REQUESTED** [review]
Sorry I had missed a couple items.

File: lib/antora/extensions/antora-linked-worktree-patch.js (hunk 1 above)
Comment (mojavelinux) [reply lib/antora/extensions/antora-linked-worktree-patch.js:3]: Not a problem.

File: lib/antora/extensions/antora-linked-worktree-patch.js (hunk 2 above)
Comment (mojavelinux) [reply lib/antora/extensions/antora-linked-worktree-patch.js:7]: Yep, sure thing.

**Pull request merged**